examples/client/client.c

```diff
@@ -4004,8 +4047,31 @@ THREAD_RETURN WOLFSSL_THREAD client_test(void* args)
 #endif /* WOLFSSL_SRTP */
 
 #ifdef WOLFSSL_TLS13
-    if (updateKeysIVs)
+    if (updateKeysIVs) {
         wolfSSL_update_keys(ssl);
+
+#ifdef WOLFSSL_DTLS13
+        if (wolfSSL_dtls(ssl) && version == -4 && waitKeyUpdate) {
+            byte foo[0];
+            while(wolfSSL_dtls_13_has_pending_msg(ssl)) {
+                err = wolfSSL_peek(ssl, foo, 0);
+                if (err < 0) {
+                    err = wolfSSL_get_error(ssl, err);
+
+                    /* non-blocking socket */
+                    if (err == WOLFSSL_ERROR_WANT_READ)
+                        continue;
+
+                    /* other peer closes the connection */
+                    if (err == WOLFSSL_ERROR_ZERO_RETURN)
+                        break;
+
+                    err_sys("wolfSSL_peek failed");
+                }
+            }
+        }
+#endif /* WOLFSSL_DTLS13 */
```

Comment:
Indeed that part of the code, besides busy looping, is broken on non-blocking sockets because it doesn't take care of DTLS timeouts. I fixed that in the last version of the code.

I still didn't implement a timeout, I think hanging tests can happen also on application data or current DTLS tests, so a more general approach is needed. I think a good solution may be to implement a general "test" timeout in `suites.c`.

Reply:
Has this section been removed?

Reply:
Yes,
because of the removal of the post-handshake authentication check (for the reason that the server doesn't have an easy way to check when post-handshake authentication succeeds), it isn't necessary anymore. It will be likely re-added in the future to enable checks that rely on the client to properly send pending messages before shutdown.